lib/rubocop/cop/style/date_time.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks for uses of 'DateTime' that should be replaced by
+      # 'Date' or 'Time'.
+      #
+      # @example
+      #
+      #   # bad - uses DateTime for current time
+      #   DateTime.now
+      #
+      #   # good - uses Time for current time
+      #   Time.now
+      #
+      #   # bad - uses DateTime for modern date
+      #   DateTime.iso8601('2016-06-29')
+      #
+      #   # good - uses Date for modern date
+      #   Date.iso8601('2016-06-29')
+      #
+      #   # good - uses DateTime with start argument for historical date
+      #   DateTime.iso8601('1751-04-23', Date::ENGLAND
+      class DateTime < Cop
+        MSG = 'Prefer Date or Time over DateTime.'.freeze
```

Comment:
Are there any rules or configuration that we could apply to make it more deterministic whether `Date` or `Time` should be used? If we are able to come up with rules or configuration, then what would you think about adding auto-correct to this cop?

Reply:
I tried my hands at this issue roughly a year ago, and did some research into this. My conclusion then was that it's somewhere between "impossible" and "not worth the effort." However I might have missed some approach that would make it easier, in case someone is keen to look into it. 🙂